src/dataflow/src/render/upsert.rs

```diff
@@ -389,58 +390,91 @@ where
                                 Some(Ok(decoded_key)) => {
                                     let decoded_value = match data.raw_data.value {
                                         None => Ok(None),
-                                        Some(value) => value.and_then(|row| {
-                                            let mut datums = Vec::with_capacity(source_arity);
-
-                                            // The datums we send to `evaluate` contain the keys
-                                            // and the values in order, so indexing works
-                                            datums.extend(decoded_key.iter());
-
-                                            datums.extend(row.iter());
-                                            datums.extend(data.metadata.iter());
-                                            evaluate(
-                                                &datums,
-                                                &predicates,
-                                                &position_or,
-                                                &mut row_packer,
-                                            )
-                                            .map_err(DataflowError::from)
-                                        }),
+                                        Some(value) => match value {
+                                            Ok(row) => {
+                                                let envelope_value = match upsert_envelope.style {
+                                                    UpsertStyle::Debezium { after_idx } => {
+                                                        match row.iter().nth(after_idx).unwrap() {
+                                                            Datum::List(after) => {
+                                                                let mut datums = Vec::with_capacity(
+                                                                    source_arity,
+                                                                );
+                                                                datums.extend(after.iter());
+                                                                Some(datums)
+                                                            }
+                                                            Datum::Null => None,
+                                                            d => panic!(
+                                                                "type error: expected record, \
+                                                                        found {:?}",
+                                                                d
+                                                            ),
+                                                        }
+                                                    }
+                                                    _ => {
```

Comment:
Can you name the other style here explicitly so that it's a compile error if we ever add more?